// @gsd/native/dist/image/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImageFormat = exports.SamplingFilter = void 0;
/** Sampling filter for resize operations. */
var SamplingFilter;
(function (SamplingFilter) {
    /** Nearest-neighbor sampling (fast, low quality). */
    SamplingFilter[SamplingFilter["Nearest"] = 1] = "Nearest";
    /** Triangle filter (linear interpolation). */
    SamplingFilter[SamplingFilter["Triangle"] = 2] = "Triangle";
    /** Catmull-Rom filter with sharper edges. */
    SamplingFilter[SamplingFilter["CatmullRom"] = 3] = "CatmullRom";
    /** Gaussian filter for smoother results. */
    SamplingFilter[SamplingFilter["Gaussian"] = 4] = "Gaussian";
    /** Lanczos3 filter for high-quality downscaling. */
    SamplingFilter[SamplingFilter["Lanczos3"] = 5] = "Lanczos3";
})(SamplingFilter || (exports.SamplingFilter = SamplingFilter = {}));
/** Output image format for encoding. */
var ImageFormat;
(function (ImageFormat) {
    /** PNG (lossless, quality ignored). */
    ImageFormat[ImageFormat["PNG"] = 0] = "PNG";
    /** JPEG (lossy, quality 0-100). */
    ImageFormat[ImageFormat["JPEG"] = 1] = "JPEG";
    /** WebP (lossless, quality ignored). */
    ImageFormat[ImageFormat["WebP"] = 2] = "WebP";
    /** GIF (quality ignored). */
    ImageFormat[ImageFormat["GIF"] = 3] = "GIF";
})(ImageFormat || (exports.ImageFormat = ImageFormat = {}));
