// @gsd/native/dist/stream-process/index.js
"use strict";
/**
 * Bash stream processor — single-pass UTF-8 decode + ANSI strip + binary sanitization.
 *
 * Handles chunk boundaries for incomplete UTF-8 and ANSI escape sequences.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.processStreamChunk = processStreamChunk;
exports.stripAnsiNative = stripAnsiNative;
exports.sanitizeBinaryOutputNative = sanitizeBinaryOutputNative;
const native_js_1 = require("../native.js");
/**
 * Process a raw bash output chunk in a single pass.
 *
 * Decodes UTF-8 (handling incomplete multibyte sequences at boundaries),
 * strips ANSI escape sequences, removes control characters (except tab and
 * newline), removes carriage returns, and filters Unicode format characters.
 *
 * Pass the returned `state` to the next call to handle sequences split
 * across chunk boundaries.
 */
function processStreamChunk(chunk, state) {
    // Convert StreamState arrays to the format napi expects (Vec<u8>)
    const napiState = state
        ? {
            utf8Pending: Array.from(state.utf8Pending),
            ansiPending: Array.from(state.ansiPending),
        }
        : undefined;
    const result = native_js_1.native.processStreamChunk(chunk, napiState);
    return {
        text: result.text,
        state: {
            utf8Pending: Array.from(result.state.utf8Pending),
            ansiPending: Array.from(result.state.ansiPending),
        },
    };
}
/**
 * Strip ANSI escape sequences from a string.
 */
function stripAnsiNative(text) {
    return native_js_1.native.stripAnsiNative(text);
}
/**
 * Remove binary garbage and control characters from a string.
 *
 * Keeps tab and newline. Removes carriage return, all other control
 * characters, Unicode format characters (U+FFF9-U+FFFB), and lone surrogates.
 */
function sanitizeBinaryOutputNative(text) {
    return native_js_1.native.sanitizeBinaryOutputNative(text);
}
