// @gsd/native/dist/text/index.js
"use strict";
/**
 * ANSI-aware text measurement and slicing.
 *
 * High-performance UTF-16 native implementation with ASCII fast-paths,
 * single-pass ANSI scanning, and proper Unicode grapheme cluster support.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EllipsisKind = void 0;
exports.wrapTextWithAnsi = wrapTextWithAnsi;
exports.truncateToWidth = truncateToWidth;
exports.sliceWithWidth = sliceWithWidth;
exports.extractSegments = extractSegments;
exports.sanitizeText = sanitizeText;
exports.visibleWidth = visibleWidth;
const native_js_1 = require("../native.js");
var types_js_1 = require("./types.js");
Object.defineProperty(exports, "EllipsisKind", { enumerable: true, get: function () { return types_js_1.EllipsisKind; } });
/**
 * Word-wrap text to a visible width, preserving ANSI escape codes across
 * line breaks.
 *
 * Active SGR codes (colors, bold, etc.) are carried to continuation lines.
 * Underline and strikethrough are reset at line ends and restored on the
 * next line.
 */
function wrapTextWithAnsi(text, width, tabWidth) {
    return native_js_1.native.wrapTextWithAnsi(text, width, tabWidth);
}
/**
 * Truncate text to a visible width with an optional ellipsis.
 *
 * @param text       Input string (may contain ANSI codes).
 * @param maxWidth   Maximum visible width in terminal cells.
 * @param ellipsisKind  0 = "\u2026", 1 = "...", 2 = none.
 * @param pad        When true, pad with spaces to exactly `maxWidth`.
 * @param tabWidth   Tab stop width (default 3, range 1-16).
 */
function truncateToWidth(text, maxWidth, ellipsisKind, pad, tabWidth) {
    return native_js_1.native.truncateToWidth(text, maxWidth, ellipsisKind, pad, tabWidth);
}
/**
 * Slice a range of visible columns from a line.
 *
 * Counts terminal cells (skipping ANSI escapes). When `strict` is true,
 * wide characters that would exceed the range are excluded.
 */
function sliceWithWidth(line, startCol, length, strict, tabWidth) {
    return native_js_1.native.sliceWithWidth(line, startCol, length, strict, tabWidth);
}
/**
 * Extract the before/after segments around an overlay region.
 *
 * ANSI state is tracked so the `after` segment renders correctly even when
 * the overlay truncates styled text.
 */
function extractSegments(line, beforeEnd, afterStart, afterLen, strictAfter, tabWidth) {
    return native_js_1.native.extractSegments(line, beforeEnd, afterStart, afterLen, strictAfter, tabWidth);
}
/**
 * Strip ANSI escape sequences, remove control characters and lone
 * surrogates, and normalize line endings (CR removed).
 *
 * Returns the original string when no changes are needed (zero-copy).
 */
function sanitizeText(text) {
    return native_js_1.native.sanitizeText(text);
}
/**
 * Calculate visible width of text excluding ANSI escape sequences.
 *
 * Tabs count as `tabWidth` cells (default 3).
 */
function visibleWidth(text, tabWidth) {
    return native_js_1.native.visibleWidth(text, tabWidth);
}
